### DISCUSSION TITLE
Passing one boolean value to one field that is set to accept only int, doesn't return error and set value to "1" or "0".

### DISCUSSION THREAD

**Discussion by Taciones** [fastapi/fastapi#9395] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Optional
from fastapi import FastAPI
from pydantic import BaseModel

class Item(BaseModel):
    name: str
    description: Optional[str] = None
    price: int
    tax: Optional[float] = None

app = FastAPI()

@app.post("/items/")
async def create_item(item: Item):
    return item
```

### Description

If we send a JSON like this:
`{
    "name": "myname",
    "price": false
}`

It returns this response:
`{
    "name": "myname",
    "description": null,
    "price": 0,
    "tax": null
}`
We can see the `"price"` turned into `0` from false value.

This is part of some negative tests we do. I thought I would receive an error code of 422.

### Operating System

macOS

### FastAPI Version

0.95.1

### Python Version

3.10.0

### Additional Context

This is my first contribution to this community. 
Please feel free to add comments or questions.

**Comment by jgould22** [comment, marked as answer]
This is expected behavior from Pydantic (the library doing this validation).

It will try and coerce the provided value to an int by calling `int(value)` on it.

If you dont want to do that you need to use the Strict types like this 
```python
class Item(BaseModel):
    name: str
    description: Optional[str] = None
    price: StrictInt
    tax: Optional[float] = None
```

**Reply by Taciones** [reply]
Awesome, it worked like a charm heree. I did some research and found this article usefull:
https://docs.pydantic.dev/usage/types/#strict-types
Thanks for the help!!
